Neos.Flow/Classes/ObjectManagement/CompileTimeObjectManager.php

```diff
@@ -211,15 +212,18 @@ protected function registerClassFiles(array $packages)
         }
 
         $availableClassNames = ['' => ['DateTime']];
-        /** @var \Neos\Flow\Package\Package $package */
+
+        $shouldRegisterFunctionalTestClasses = (bool)($this->allSettings['Neos']['Flow']['object']['registerFunctionalTestClasses'] ?? false);
+
+        /** @var \Neos\Flow\Package\PackageInterface $package */
         foreach ($packages as $packageKey => $package) {
-            if ($package->isObjectManagementEnabled() && (ComposerUtility::isFlowPackageType($package->getComposerManifest('type')) || isset($includeClassesConfiguration[$packageKey]))) {
+            if (ComposerUtility::isFlowPackageType($package->getComposerManifest('type')) || isset($includeClassesConfiguration[$packageKey])) {
                 foreach ($package->getClassFiles() as $fullClassName => $path) {
                     if (substr($fullClassName, -9, 9) !== 'Exception') {
                         $availableClassNames[$packageKey][] = $fullClassName;
                     }
                 }
-                if (isset($this->allSettings['Neos']['Flow']['object']['registerFunctionalTestClasses']) && $this->allSettings['Neos']['Flow']['object']['registerFunctionalTestClasses'] === true) {
+                if ($package instanceof FlowPackageInterface && $shouldRegisterFunctionalTestClasses) {
                     foreach ($package->getFunctionalTestsClassFiles() as $fullClassName => $path) {
```

Comment:
`getFunctionalTestsClassFiles()` is no longer part of the `PackageInterface`

Reply:
`if ($package instanceof FlowPackageInterface && `

Reply:
Added it to the FlowPackageInterface for now as `@internal`